### DISCUSSION TITLE
Microphone found in GNOME, but not in i3

### DISCUSSION THREAD

**Discussion by SpangeJ** [i3/i3#5728] [Q&A]
I have a similar [issue#3264](https://github.com/i3/i3/issues/3264)

I'm running version `4.20.1` of i3

## I'm submitting a…
<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
I have a Bluetooth headset which I have paired and connected with using `bluetoothctl`
I can listen to music but i3 cannot find its microphone in either `pavucontrol` or in Slack.
However, if I log out of i3 and log in with UBUNTU22.04/GNOME I can select the microphone there.
That is why I think this is an i3 bug.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.20.1 (2021-11-03) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.20.1 (2021-11-03) (pid 2211)
Loaded i3 config:
  /home/user/.config/i3/config (main) (last modified: fr. 20. okt. 2023 kl. 10.37 +0200, 2466 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

`uname -a`:
`Linux machineName 6.2.0-34-generic #34~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 13:12:03 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`

P.S.
I do not know how I could log this in a log file, any guidance there is much appreciated.

**Comment by orestisfl** [comment]
i3 has nothing to do with your sound system, there is no code in i3 to interact with pulseaudio, alsa or pipewire. Gnome might be configuring something that enables you to use your microphone, you need to find what that missing part is and also use it in your i3 session.

**Reply by SpangeJ** [reply]
Thank you @orestisfl any clue as to how I might find the missing part?

**Reply by orestisfl** [reply]
I don't use Ubuntu 22.04 (which I think still defaults to pulseaudio) so I don't really know but you could take a look in https://wiki.archlinux.org/title/bluetooth_headset and see if you find something that helps. Keep in mind that there might be difference between what the article says and what your system does but it could be a good starting point.

**Reply by SpangeJ** [reply]
Thank you @orestisfl 
I found a (not-so-good) solution 🎉
I launched `pavucontrol` and under `Configuration` I found:
![image](https://github.com/i3/i3/assets/21217166/15022740-2066-4978-ab90-75214b9b2153)
If I changed it to:
![image](https://github.com/i3/i3/assets/21217166/647c9544-90df-493e-aaab-a33e4d317022)
I now can see my mic:
![image](https://github.com/i3/i3/assets/21217166/f0c6f7f7-5c38-49d4-8126-b1fc56575416)
![image](https://github.com/i3/i3/assets/21217166/67d0b4e8-1dfd-4d9a-8dc7-5411d9f0869e)

....

However, when I select `Handsfree Head Unit (HFP)` the sound is MONO and really poor (sounds far away, flat, lower fidelity) compared to `High Fidelity Playback (A2SP Sink), I'll see if I can get the best of two worlds! But at least I figured something out.

**Reply by SpangeJ** [reply]
Update, I replaced [pulseaudio with pipewire](https://ubuntuhandbook.org/index.php/2022/04/pipewire-replace-pulseaudio-ubuntu-2204/). Which helped a lot. Now I have a hotkey binding for swapping between settings:
`pactl set-card-profile bluez_card.00_16_94_7E_CC_1C a2dp-sink-sbc_xq` selects a high fidelity setting, but the microphone is disabled.
`pactl set-card-profile bluez_card.00_16_94_7E_CC_1C headset-head-unit` enabled the mic, but changed audio settings to poor.
I've added these two commands to hotkeys, so whenever I'm called on Slack or am about to enter a meeting I switch to `headset-head-unit`

However; I am still not really pleased, it turns out that in `a2dp-sink-sbc_xq` the sound transfer might glitch causing the sound to be distorted for several seconds or just plainly gone. Always lasts for about 5s, the frequency is from several times pr. minute to once every fifth minute, so pretty annoying, hard to get a good zone focus for developing.

**Comment by SpangeJ** [comment]
The @i3bot wrote:
> Sorry, we can only support the latest major version. Please upgrade from 4.20 to 4.22, verify the bug still exists, and re-open this issue.

Any documentation on how I do that upgrade?
I do get a bit nervous, after reading [this on Reddit](https://www.reddit.com/r/i3wm/comments/10l8exu/trying_to_update_from_4201_to_422_on_a_linux_mint/)

**Reply by SpangeJ** [reply]
https://forums.linuxmint.com/viewtopic.php?t=388673
This should work

**Reply by orestisfl** [reply]
The bot's comment was regarding your original bug issue. Since i3 is not really involved here, you don't need to upgrade. Of course, you could still do it if you need a new feature (e.g. gaps).
